src/libraries/System.IO.Compression/src/System/IO/Compression/ZLibCompressionOptions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.IO.Compression
+{
+    /// <summary>
+    /// Provides compression options to be used with <see cref="ZLibStream"/>, <see cref="DeflateStream"/> and <see cref="GZipStream"/>.
+    /// </summary>
+    public sealed class ZLibCompressionOptions
+    {
+        private int _compressionLevel = -1;
+        private ZLibCompressionStrategy _strategy;
+
+        /// <summary>
+        /// Gets or sets the compression level for a compression stream.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when the value is less than -1 or greater than 9.</exception>"
+        /// <remarks>
+        /// Can accept any value between -1 and 9 (inclusive), 0 gives no compression,  1 gives best speed, 9 gives best compression.
+        /// and -1 requests the default compression level which is currently equivalent to 6.
+        /// </remarks>
+        public int CompressionLevel
+        {
+            get => _compressionLevel;
+            set
+            {
+                ArgumentOutOfRangeException.ThrowIfLessThan(value, -1);
+                ArgumentOutOfRangeException.ThrowIfGreaterThan(value, 9);
+
+                _compressionLevel = value;
+            }
+        }
+        /// <summary>
+        /// Gets or sets the compression algorithm for a compression stream.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException" >Thrown when the value is not a valid <see cref="ZLibCompressionStrategy"/> value.</exception>"
```

Comment:
Exceptions should read as if you were to implicitly complete the sentence "_The exception that is thrown when..._":
```suggestion
        /// <exception cref="ArgumentOutOfRangeException" >The value is not a valid <see cref="ZLibCompressionStrategy"/> value.</exception>
```